vector/src/main/java/im/vector/app/features/link/LinkHandlerActivity.kt

```diff
@@ -55,19 +60,56 @@ class LinkHandlerActivity : VectorBaseActivity() {
         }
 
         if (uri.path == PATH_CONFIG) {
-            if (sessionHolder.hasActiveSession()) {
-                displayAlreadyLoginPopup(uri)
-            } else {
-                // user is not yet logged in, this is the nominal case
-                startLoginActivity(uri)
-            }
+            handleConfigUrl(uri)
+        } else if (SUPPORTED_HOSTS.contains(uri.host)) {
+            handleSupportedHostUrl(uri)
+        }
+    }
+
+    private fun handleConfigUrl(uri: Uri) {
+        if (sessionHolder.hasActiveSession()) {
+            displayAlreadyLoginPopup(uri)
         } else {
-            // Other link are not yet handled, but should not comes here (manifest configuration error?)
-            Timber.w("Unable to handle this uir: $uri")
+            // user is not yet logged in, this is the nominal case
+            startLoginActivity(uri)
+        }
+    }
+
+    private fun handleSupportedHostUrl(uri: Uri) {
+        if (!sessionHolder.hasActiveSession()) {
+            startLoginActivity(uri)
             finish()
+        } else {
+            convertUriToPermalink(uri)?.let { permalink ->
+                startPermalinkHandler(permalink)
+            } ?: run {
+                // Host is correct but we do not recognize path
+                Timber.w("Unable to handle this uri: $uri")
+                finish()
+            }
         }
     }
 
+    private fun convertUriToPermalink(uri: Uri): String? {
+        val path = SUPPORTED_PATHS.find { it in uri.toString() } ?: return null
```

Comment:
You should rather test if `uri.path` starts with one of the supported paths.

Reply:
Unfortunately Android cannot parse path because of "#" in the url.

Reply:
Ok